src/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.MappingTables.cs

```diff
@@ -343,7 +343,16 @@ public unsafe sealed override bool TryGetConstructedGenericTypeForComponents(Run
 
             TypeInfo[] typeArguments = new TypeInfo[genericTypeArgumentHandles.Length];
             for (int i = 0; i < genericTypeArgumentHandles.Length; i++)
+            {
+                // Early out if one of the arguments is a generic definition.
+                // The reflection stack will use this to construct a Type that doesn't have a type handle.
+                // Note: this is different from the validation we do in EnsureSatisfiesClassConstraints because this
+                // should not throw.
+                if (RuntimeAugments.IsGenericTypeDefinition(genericTypeArgumentHandles[i]))
```

Comment:
Agreed - this is a low-level method that should return a boolean (or Debug.Assert if truly nonsensical input like a null array) but it shouldn't throw exceptions for "bad input."